src/components/templates/product-table/index.tsx

```diff
@@ -0,0 +1,614 @@
+        navigate(`/a/products/${product.id}`)
+        toaster("Created a new return reason", "success")
+      },
+      onError: (error) => {
+        toaster(getErrorMessage(error), "error")
+      },
+    })
+  }
+
+  const getActionablesForProduct = (product) => {
+    return [
+      {
+        label: "Edit",
+        onClick: () => navigate(`/a/products/${product.id}`),
+        icon: <EditIcon size={20} />,
+      },
+      {
+        label: product.status === "published" ? "Unpublish" : "Publish",
+        onClick: () => {
+          setUpdateProduct(product)
+          updateProductHook.mutate(
+            {
+              status: product.status === "published" ? "draft" : "published",
+            },
+            {
+              onSuccess: () => {
+                toaster(
+                  `Successfully ${
+                    product.status === "published" ? "unpublished" : "published"
+                  } product`,
+                  "success"
+                )
+                refetch()
+              },
+              onError: (err) => toaster(getErrorMessage(err), "error"),
+            }
+          )
+          // Medusa.products
+          //   .update(product.id, {
+          //     status: product.status === "published" ? "draft" : "published",
+          //   })
+          //   .then(() =>
+          //     toaster(
+          //       `Successfully ${
+          //         product.status === "published" ? "unpublished" : "published"
+          //       } product`,
+          //       "success"
+          //     )
+          //   )
+          //   .then(() => refetch())
+          //   .catch((err) => toaster(getErrorMessage(err), "error"))
+        },
+        icon:
+          product.status === "published" ? (
+            <UnpublishIcon size={20} />
+          ) : (
+            <PublishIcon size={20} />
+          ),
+      },
+      {
+        label: "Duplicate",
+        onClick: () => handleCopyProduct(product),
+        icon: <DuplicateIcon size={20} />,
+      },
+      {
+        label: "Delete",
+        variant: "danger",
+        onClick: () => setDeleteProduct(product),
+        icon: <TrashIcon size={20} />,
+      },
+    ]
+  }
+
+  return (
+    <div className="w-full h-full overflow-y-scroll">
+      {isLoading || isRefetching || !products ? (
+        <div className="w-full pt-2xlarge flex items-center justify-center">
+          <Spinner size={"large"} variant={"secondary"} />
+        </div>
+      ) : (
+        <>
+          <Table
+            filteringOptions={
+              <ProductsFilter
+                setStatusFilter={setStatusFilter}
+                statusFilter={statusFilter}
+                setCollectionFilter={setCollectionFilter}
+                collectionFilter={collectionFilter}
+                collections={collectionsList}
+                setTagsFilter={toggleFilterTags}
+                submitFilters={submitFilters}
+                tagsFilter={tagsFilter}
```

Comment:
Do you think we can do something to avoid all this? I'm thinking `useReducer` might help here .. you would then pass on your dispatch function + filters object to the `ProductsFilter`. I'm thinking the state/filters could be something like this:
```javascript
{
  "status": { open: false,  value: "" },
  "collections": { open: false, value: "" },
  ...
}
```

Just an initial thought, main idea here is to do something to make the `ProductTable` a bit simpler and shorter